checker/src/abstract_value.rs

```diff
@@ -2017,6 +1997,31 @@ impl AbstractValueTrait for Rc<AbstractValue> {
         self.expression.record_heap_blocks(result);
     }
 
+    /// True if the value is derived from one or more memory locations whose addresses were not known
+    /// when the value was constructed. Refining such values in the current environment could
+    /// resolve them to particular locations and those locations may have more useful associated values.
+    #[logfn_inputs(TRACE)]
+    fn refers_to_unknown_location(&self) -> bool {
+        match &self.expression {
+            Expression::Cast { operand, .. } => operand.refers_to_unknown_location(),
+            Expression::ConditionalExpression {
+                consequent,
+                alternate,
+                ..
+            } => consequent.refers_to_unknown_location() || alternate.refers_to_unknown_location(),
+            Expression::Reference(..) => true,
+            Expression::UninterpretedCall { path, .. }
+            | Expression::Variable { path, .. }
+            | Expression::Widen { path, .. } => {
+                if let PathEnum::Alias { value } = &path.value {
+                    return value.refers_to_unknown_location();
+                }
+                false
+            }
+            _ => false,
```

Comment:
Does this function also need to consider `Join` and `Switch` expressions, since they are similar to `ConditionalExpression`?

Reply:
Excellent question. I think the answer is yes.